data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -67,61 +69,86 @@ public PublishedNewsDisplayedPropUpgrade(InitParams initParams,
+        processedPublishedNewsCount += processedPublishedNewsCountByTransaction;
+        migratedPublishedNewsCount += processedPublishedNewsCountByTransaction - notMigratedPublishedNewsCountByTransaction;
+        notMigratedPublishedNewsCount += notMigratedPublishedNewsCountByTransaction; 
+        LOG.info("Published news migration progress: processed={}/{} succeeded={} error={}", processedPublishedNewsCountByTransaction, processedPublishedNewsCount, migratedPublishedNewsCount, notMigratedPublishedNewsCount);
+      } catch (Exception e) {
+        throw new IllegalStateException ("Error when managing news target metadata items props", e);
+      } finally {
+        ExoContainerContext.setCurrentContainer(null);
       }
-
-      for (MetadataItemEntity newsTargetsMetadataItem : newsTargetsMetadataItems) {
-        StringBuilder deleteNewsTargetsMetadataItemsPropsQueryStringBuilder = new StringBuilder("DELETE FROM SOC_METADATA_ITEMS_PROPERTIES WHERE METADATA_ITEM_ID = '");
-        deleteNewsTargetsMetadataItemsPropsQueryStringBuilder.append(newsTargetsMetadataItem.getId() + "'");
-        deleteNewsTargetsMetadataItemsPropsQueryStringBuilder.append(" AND (NAME = '" + STAGED_STATUS + "' OR NAME = '" + NewsUtils.DISPLAYED_STATUS + "')");
-        Query deleteNewsTargetsMetadataItemsPropsQuery = entityManager.createNativeQuery(deleteNewsTargetsMetadataItemsPropsQueryStringBuilder.toString(), MetadataItemEntity.class);
-        deleteNewsTargetsMetadataItemsPropsQuery.executeUpdate();
+    }
+    LOG.info("End published news migration: total={} succeeded={} error={}. It tooks {} ms.", totalPublishedNewsCount, migratedPublishedNewsCount, notMigratedPublishedNewsCount, (System.currentTimeMillis() - startupTime));
+  }
+  
+  @ExoTransactional
+  public List<MetadataItemEntity> getNewsTargetMetadataItems() throws IllegalStateException {
+    List<String> newsTargetMetadatas = metadataService.getMetadatas(NewsTargetingService.METADATA_TYPE.getName(), 0).stream()
+        .map(newsTargetMetadata -> String.valueOf(newsTargetMetadata.getId()))
+        .collect(Collectors.toList());
+    List<MetadataItemEntity> newsTargetsMetadataItems = new ArrayList<>();
+    EntityManager entityManager = entityManagerService.getEntityManager();
+    try {
+      Query getNewsTargetMetadataItemsQuery = entityManager.createNativeQuery("SELECT * FROM SOC_METADATA_ITEMS WHERE METADATA_ID IN :newsTargetMetadatas", MetadataItemEntity.class);
+      getNewsTargetMetadataItemsQuery.setParameter("newsTargetMetadatas", newsTargetMetadatas);
+      newsTargetsMetadataItems = getNewsTargetMetadataItemsQuery.getResultList();
+    } catch (Exception e) {
+      throw new IllegalStateException("Error when getting news target metadata items", e);
+    }
+    return newsTargetsMetadataItems;
+  }
+  
+  @ExoTransactional
+  public int manageNewsTargetsMetadataItemsProps(List<MetadataItemEntity> newsTargetsMetadataItems) throws IllegalStateException {
```

Comment:
```suggestion
  public int manageNewsTargetsMetadataItemsProps(List<MetadataItemEntity> newsTargetsMetadataItems) {
```
No need to add throws for unchecked exceptions